Mathlib/Analysis/Meromorphic/NormalFormAt.lean

```diff
@@ -135,6 +124,151 @@ theorem AnalyticAt.meromorphicNFAt (hf : AnalyticAt 𝕜 f x) :
     MeromorphicNFAt f x := by
   simp [meromorphicNFAt_iff_analyticAt_or, hf]
 
+/-- Meromorphic functions have normal form outside of a discrete subset in the domain of
+meromorphicity. -/
+theorem MeromorphicOn.meromorphicNFAt_codiscreteWithin [CompleteSpace E] {U : Set 𝕜}
+    (hf : MeromorphicOn f U) :
+    { x | MeromorphicNFAt f x } ∈ Filter.codiscreteWithin U := by
+  filter_upwards [hf.analyticAt_codiscreteWithin] with _ ha
+  exact ha.meromorphicNFAt
+
+/- Private helper lemma. -/
+private lemma WithTop.map_natCast_eq_zero {n : WithTop ℕ}
+  (hn : WithTop.map (Nat.cast : ℕ → ℤ) n = 0) :
+  n = 0 := by
+  rcases n
+  · contradiction
+  · simp only [WithTop.map, Option.map] at hn
+    rw [Int.ofNat_eq_zero.mp (WithTop.coe_eq_zero.mp hn)]
+    rfl
+
+/-!
+## Vanishing and order
+-/
+
+/-- If `f` is meromorphic in normal form at `z₀`, then `f` has order zero iff it does not vanish at
+`z₀`.
+
+See `AnalyticAt.order_eq_zero_iff` for an analogous statement about analytic functions. -/
+theorem MeromorphicNFAt.order_eq_zero_iff {f : 𝕜 → E} {x : 𝕜} (hf : MeromorphicNFAt f x) :
+    hf.meromorphicAt.order = 0 ↔ f x ≠ 0 := by
+  constructor
+  · intro h₁f
+    have h₂f := hf.order_nonneg_iff_analyticAt.1 (le_of_eq h₁f.symm)
+    apply h₂f.order_eq_zero_iff.1
+    apply WithTop.map_natCast_eq_zero
+    rwa [h₂f.meromorphicAt_order] at h₁f
+  · intro h
+    have hf' := hf
+    rcases hf with h₁ | ⟨n, g, h₁g, h₂g, h₃g⟩
+    · have := h₁.eq_of_nhds
+      tauto
+    · have : n = 0 := by
+        by_contra hContra
+        have := Filter.EventuallyEq.eq_of_nhds h₃g
+        simp only [Pi.smul_apply', Pi.pow_apply, sub_self, zero_zpow n hContra, zero_smul] at this
+        tauto
+      simp only [this, zpow_zero, smul_eq_mul, one_mul] at h₃g
+      apply (hf'.meromorphicAt.order_eq_int_iff 0).2
+      use g, h₁g, h₂g
+      simp only [zpow_zero, smul_eq_mul, one_mul]
+      exact h₃g.filter_mono nhdsWithin_le_nhds
+
+/-!
+## Local nature of the definition and local identity theorem
+-/
+
+/-- Local identity theorem: two meromorphic functions in normal form agree in a
+neighborhood iff they agree in a pointed neighborhood.
+
+See `AnalytivAt.eventuallyEq_nhdNE_iff_eventuallyEq_nhd` for the analogous
+statement for analytic functions.
+-/
+theorem MeromorphicNFAt.eventuallyEq_nhdNE_iff_eventuallyEq_nhd {f g : 𝕜 → E} {z₀ : 𝕜}
```

Comment:
I think `f`, `g`, and `z₀` can be turned into `variable`s. The case where you need `f : 𝕜 → 𝕜` could either be done by shadowing `f` in the statement, or by also adding a variable `phi`(?) of that type.

Reply:
Implemented.